staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -0,0 +1,290 @@
+			recordEvent(eventBroadcaster.sink, event, true)
+		}
+
+	}
+}
+
+func (eventBroadcaster *eventBroadcasterImpl) checkEndingSeries() {
+	eventBroadcaster.mu.Lock()
+	defer eventBroadcaster.mu.Unlock()
+	for isomorphicKey, event := range eventBroadcaster.eventCache {
+		eventSerie := event.Series
+		if eventSerie != nil {
+
+			if eventSerie.LastObservedTime.Time.Add(6 * time.Minute).Before(time.Now()) {
+
+				if recordEvent(eventBroadcaster.sink, event, true) {
+					delete(eventBroadcaster.eventCache, isomorphicKey)
+				}
+			}
+		}
+
+	}
+}
+
+// NewRecorder returns an EventRecorder that records events with the given event source.
+func (eventBroadcaster *eventBroadcasterImpl) NewRecorder(scheme *runtime.Scheme, reportingController string) EventRecorder {
+
+	hostname, _ := os.Hostname()
+	reportingInstance := reportingController + "-" + hostname
+
+	return &recorderImpl{scheme, reportingController, reportingInstance, eventBroadcaster.Broadcaster, clock.RealClock{}}
+}
+
+func (eventBroadcaster *eventBroadcasterImpl) recordToSink(event *v1beta1.Event, randGen *rand.Rand, clock clock.Clock) {
+	// Make a copy before modification, because there could be multiple listeners.
+	// Events are safe to copy like this.
+	eventCopy := *event
+	event = &eventCopy
+
+	go func() {
+		eventKey := getKey(event)
+		eventBroadcaster.mu.Lock()
+		defer eventBroadcaster.mu.Unlock()
+		if isomorphicEvent, isIsomorphic := eventBroadcaster.eventCache[eventKey]; isIsomorphic {
+
+			if isomorphicEvent.Series != nil {
+				isomorphicEvent.Series.Count++
+				isomorphicEvent.EventTime = metav1.MicroTime{Time: clock.Now()}
+
+			} else {
+				isomorphicEvent.Series = &v1beta1.EventSeries{
+					Count:            1,
+					LastObservedTime: metav1.MicroTime{Time: clock.Now()},
+				}
+				attemptRecording(eventBroadcaster.sink, isomorphicEvent, true, randGen, eventBroadcaster.sleepDuration)
+
+			}
+
+		} else {
+			eventBroadcaster.eventCache[eventKey] = event
+			attemptRecording(eventBroadcaster.sink, event, false, randGen, eventBroadcaster.sleepDuration)
+		}
+
+	}()
+
+}
+
+func attemptRecording(sink EventSink, event *v1beta1.Event, b bool, randGen *rand.Rand, sleepDuration time.Duration) {
+	tries := 0
+
+	for {
+		if recordEvent(sink, event, b) {
+			break
+		}
+		tries++
+		if tries >= maxTriesPerEvent {
+			klog.Errorf("Unable to write event '%#v' (retry limit exceeded!)", event)
+			break
+		}
+		// Randomize the first sleep so that various clients won't all be
+		// synced up if the master goes down.
+		if tries == 1 {
+			time.Sleep(time.Duration(float64(sleepDuration) * randGen.Float64()))
+		} else {
+			time.Sleep(sleepDuration)
+		}
+	}
+
+}
+
+func recordEvent(sink EventSink, event *v1beta1.Event, isEventSerie bool) bool {
+	var newEvent *v1beta1.Event
+	var err error
+	if isEventSerie {
+
+		patch, err := createPatchBytes(event)
+		if err != nil {
+			klog.Errorf("Unable to calculate diff, no merge is possible: %v", err)
+			return false
```

Comment:
Retrying this error probably won't help at all - so we should probably return true.